test/test_ourdna_dashboard.py

```diff
@@ -0,0 +1,397 @@
+from collections import OrderedDict
+from datetime import datetime
+from math import ceil
+from test.testbase import DbIsolatedTest, run_as_sync
+
+from db.python.layers.ourdna.dashboard import OurDnaDashboardLayer
+from db.python.layers.participant import ParticipantLayer
+from db.python.layers.sample import SampleLayer
+from db.python.tables.sample import SampleFilter
+from db.python.utils import GenericFilter
+from models.models import (
+    ParticipantUpsert,
+    ParticipantUpsertInternal,
+    SampleUpsert,
+    SampleUpsertInternal,
+)
+
+
+def str_to_datetime(timestamp_str):
+    """Convert string timestamp to datetime"""
+    return datetime.strptime(timestamp_str, '%Y-%m-%d %H:%M:%S')
+
+
+class OurDNADashboardTest(DbIsolatedTest):
+    """Test sample class"""
+
+    @run_as_sync
+    async def setUp(self) -> None:
+        # don't need to await because it's tagged @run_as_sync
+        super().setUp()
+        self.odd = OurDnaDashboardLayer(self.connection)
+        self.sl = SampleLayer(self.connection)
+        self.pl = ParticipantLayer(self.connection)
+
+        participants = await self.pl.upsert_participants(
+            [
```

Comment:
Can you move these static variables to be defined as constants to make it easier to scan the setup function